proxyd/methods.go

```diff
@@ -20,366 +16,27 @@ type RPCMethodHandler interface {
 }
 
 type StaticMethodHandler struct {
-	cache interface{}
+	cache Cache
 	m     sync.RWMutex
 }
 
-func (e *StaticMethodHandler) GetRPCMethod(ctx context.Context, req *RPCReq) (*RPCRes, error) {
-	e.m.RLock()
-	cache := e.cache
-	e.m.RUnlock()
-
-	if cache == nil {
-		return nil, nil
-	}
-	return &RPCRes{
-		JSONRPC: req.JSONRPC,
-		Result:  cache,
-		ID:      req.ID,
-	}, nil
+func (e *StaticMethodHandler) key(req *RPCReq) string {
+	// signature is a cache-friendly base64-encoded string with json.RawMessage param contents
+	signature := base64.StdEncoding.EncodeToString(req.Params)
```

Comment:
Can this be a hash function of some kind instead? I'm concerned about putting raw request params (which are untrusted user input, and unbounded in size) into the cache. Could use something simple like SHA256.